src/autorest-core/lib/autorest-core.ts

```diff
@@ -17,19 +17,20 @@ export class AutoRest extends EventEmitter {
   private _configurations = new Array<any>();
   private _view: ConfigurationView | undefined;
   public get view(): Promise<ConfigurationView> {
-    return (async () => {
-      if (!this._view) {
-        const messageEmitter = new MessageEmitter();
+    return (this._view) ? Promise.resolve(this._view) : this.RegenerateView(true);
+  }
 
-        // subscribe to the events for the current configuration view
-        messageEmitter.GeneratedFile.Subscribe((cfg, file) => this.GeneratedFile.Dispatch(file));
-        messageEmitter.Message.Subscribe((cfg, message) => this.Message.Dispatch(message));
+  public async RegenerateView(includeDefault: boolean = false): Promise<ConfigurationView> {
```

Comment:
this makes it possible to get the view without the default document included. Nice for batch mode.